### ISSUE TITLE
Activity created but not show up

### ISSUE THREAD

**Issue by ibrahimKonyali** [istornz/flutter_live_activities#93]
### Hello all,

I'm following the procedure and can't get the activity show up on lock screen.

**Here what i did so far;**

Opened; ios/Runner.xcworkspace
Added widget extension with embed application runner is selected.
Activated the widget extension from dialog
Added push notification to runner
Enabled NSSupportsLiveActivities for both widget extension info and runner info
Added App group capability for both widget extension and runner and gave same id for both
I used example code for native side and gave same app group id to UserDefaults

[https://github.com/istornz/flutter_live_activities/blob/main/example/ios/extension-example/extension_example.swift](url)

`let sharedDefault = UserDefaults(suiteName: "YOUR_GROUP_ID")!`

And on flutter side, i init() the plugin and call createActivity()

After doing all of these and tried to create activity, this is only error i get from runner (_it creates activity and return activity id without problem_)

`[User Defaults] Couldn't read values in CFPrefsPlistSource<0x28010f900> (Domain: group.xxx.xxx, User: kCFPreferencesAnyUser, ByHost: Yes, Container: (null), Contents Need Refresh: Yes): Using kCFPreferencesAnyUser with a container is only allowed for System Containers, detaching from cfprefsd`

Even id in app group for both in extension and runner is same and that id is active on Certificates, Identifiers & Profiles at developer at apple it still doesn't show the activity on lock screen.

**If you are going to suggest using app group id with TeamID in it, i already tried that and didn't work either.**

**Comment by DKMonster** [comment]
Could you share your Swift code?

I encountered the same issues before, but I managed to fix them.

I think we're running into the same problem.

**Comment by DKMonster** [comment]
When implementing the Live Activity widget:

1. The top-level attributes struct must be named `LiveActivitiesAppAttributes` in your Swift code.
   This is essential because the Flutter method channel uses this specific type to communicate with the widget.

2. The widget struct itself (below the attributes) can have any name you choose.

If your widget isn't displaying, please check if you've properly implemented the `LiveActivitiesAppAttributes` struct.

**Comment by ibrahimKonyali** [comment]
> When implementing the Live Activity widget:
> 
> 1. The top-level attributes struct must be named `LiveActivitiesAppAttributes` in your Swift code.
>    This is essential because the Flutter method channel uses this specific type to communicate with the widget.
> 2. The widget struct itself (below the attributes) can have any name you choose.
> 
> If your widget isn't displaying, please check if you've properly implemented the `LiveActivitiesAppAttributes` struct.

Yeah i use same struct with example but didn't work. Then i use this package [https://pub.dev/packages/flutter_live_activities](url) and without app group and custom struct i was able to create and update activities after i uncheck "copy only when installing" under the embed frameworks in build phases.

**Comment by manoj1749** [comment]
Heyy @ibrahimKonyali @DKMonster I am also having the same issue, my flutter engine is updating and creating the activity, but the live activity, notification and dynamic island are not showing, in both my simulator and mobile. But when the app is in background and when I click on Dynamic Island it's opening the app.
This is my Swift code
```swift
//
//  extension_example.swift
//  extension-example
//
//  Created by Dimitri Dessus on 28/09/2022.
//

import ActivityKit
import WidgetKit
import SwiftUI

@main
struct Widgets: WidgetBundle {
  var body: some Widget {
    if #available(iOS 16.1, *) {
      RideTrackingApp()
    }
  }
}

struct LiveActivitiesAppAttributes: ActivityAttributes, Identifiable {
    public typealias RideData = ContentState

    public struct ContentState: Codable, Hashable {
              var rideStatus: String
        var elapsedTime: String
        var distanceCovered: String
    }

    var id = UUID()
}

let sharedDefault = UserDefaults(suiteName: "group.app.liveactivities")!

@available(iOSApplicationExtension 16.1, *)
struct RideTrackingApp: Widget {
  var body: some WidgetConfiguration {
    ActivityConfiguration(for: LiveActivitiesAppAttributes.self) { context in
//     let pizzaName = sharedDefault.string(forKey: context.attributes.prefixedKey("rideStatus"))! // put the same key as your Dart map
// let pizzaPrice = sharedDefault.string(forKey: context.attributes.prefixedKey("elapsedTime"))
// let quantity = sharedDefault.string(forKey: context.attributes.prefixedKey("distanceCovered"))
      VStack(alignment: .leading) {
        Text(context.state.rideStatus)
          .font(.headline)
          .padding(.bottom, 4)
        
        HStack {
          Text("Time:")
            .font(.caption)
          Text(context.state.elapsedTime)
            .font(.body)
            .foregroundColor(.blue)
        }
        
        HStack {
          Text("Distance:")
            .font(.caption)
          Text(context.state.distanceCovered)
            .font(.body)
            .foregroundColor(.green)
        }
      }
      .activityBackgroundTint(Color.gray.opacity(0.2))
      .activitySystemActionForegroundColor(Color.blue)
    } dynamicIsland: { context in
      DynamicIsland {
        DynamicIslandExpandedRegion(.leading) {
          VStack {
            Text("Status")
            Text(context.state.rideStatus)
              .font(.headline)
              .foregroundColor(.primary)
          }
        }
        
        DynamicIslandExpandedRegion(.trailing) {
          VStack {
            Text("Time")
            Text(context.state.elapsedTime)
              .font(.headline)
              .foregroundColor(.primary)
          }
        }
        
        DynamicIslandExpandedRegion(.bottom) {
          VStack {
            Text("Distance")
            Text(context.state.distanceCovered)
              .font(.headline)
              .foregroundColor(.primary)
          }
        }
      } compactLeading: {
        Text("T")
      } compactTrailing: {
        Text("D")
      } minimal: {
        Text("R")
      }
      .widgetURL(URL(string: "http://www.apple.com"))
      .keylineTint(Color.blue)
    }
  }
}

extension LiveActivitiesAppAttributes {
  func prefixedKey(_ key: String) -> String {
    return "\(id)_\(key)"
  }
}

```

**Comment by thpengu** [comment]
Hello everyone,

I'm trying to integrate Dynamic Island functionality into my Flutter app using the package live_activities, and I've followed the documentation thoroughly. However, I'm running into some issues. Specifically, Dynamic Island does not appear on the screen. I tested a real device that supports this function.

@ibrahimKonyali @manoj1749, have you encountered this issue or found a solution? I would appreciate any advice or guidance on how to make Dynamic Island work in Flutter.

I appreciate any help you can provide.

**Comment by manoj1749** [comment]
What error is it showing for you?
@thpengu

**Comment by thpengu** [comment]
@manoj1749 
There is no error showing. I checked, and the live activity is created but not appearing on my screen. I tested the real device and the emulator, too. I think it's the same issue as yours. 

<But when the app is in background and when I click on Dynamic Island it's opening the app.>
In my case, it opens only on the emulator if I click the dynamic island area.

**Comment by manoj1749** [comment]
You can see the dynamic island minimising and maximising when holded and all?. If that so, can you check whether the data which you are trying to send form flutter is being sent to native side or not? Also share youe AppDelegate.swift of the widget.

**Comment by RahnRazamai** [comment]
are you using flutter flavor? @thpengu i having the same issue with this library. my guess is because my staging flavor having different bundle id from the prod one. and the live activities is attached to prod bundle id

**Comment by sbergmair** [comment]
In my case live activities did not show up after upgrading to `flutter: 3.27.1`. The changes on `Color` in dart broke the live activities, because deserialization did not work any more. Fixing this also fixed the live activities. Maybe this helps anyone.

**Comment by byunggul-kim** [comment]
I recommend check MyAppWidgetExtension.appex is added Frameworks, Libraries, and Embedded Content.

![Image](https://github.com/user-attachments/assets/35a2bf1b-3f7a-4b0a-b1ce-2cdb2d4ab4f5)

**Comment by Azzeccagarbugli** [comment]
For me unchecking the _"Copy only when installing"_ check in the _"Embedded Foundation Extension"_ in the build phase of the target runner made the live activity work again ✨

**Comment by TarekAlabd** [comment]
The same issue here
Did you find any solution?
@ibrahimKonyali @manoj1749

**Comment by manoj1749** [comment]
Yeah, It did worked out after a while, after doing some debugging and few other trail and error methods. I had worked on it while I am interning at a company.

**Comment by itsatishay** [comment]
In my implementation i have noticed the same issue. Not able to figure out what is the issue. 

But did notice one thing that if i install the app again on ios simulator it works sometimes. It is almost at random. So i just keep installing again and again till eventually it starts working. 
Similar issue when testing on real device.

**Comment by Mohamed-Ahmed-Salah** [comment]
I was facing the same issue.  

You're not using User Defaults your passing and using variables from ContentState. 
which when its required and not nullable will cause the issue of the activity not showing. 
Because the package is not passing the contentState and is using UserDefault which make the variables in contentState always null. 
example: 
 public struct ContentState: Codable, Hashable {
        var status: String?
    }

 let state = context.state
 Text("status: " + (state.status ?? "NOT FOUND"))
from flutter: 
 _currentActivityId = await _plugin.createActivity(
        'a

_${DateTime.now().millisecondsSinceEpoch}',
        {
            "status":status
        },
        removeWhenAppIsKilled: false,
      );
For me always returns not found... 


for correct Implementation as per the documentation it's using User Defaults
so in flutter I'm sending data using HomeWidget
HomeWidget.saveWidgetData<String>(
        prefixedKey('status'),
        data['status'] as String,
      ),

so first remove the variables from the contentState (this will not allow server update and you only rely on updates from flutter and app being not terminated)

define sharedDefault 
private let sharedDefault = UserDefaults(suiteName: "group.com.companyNamye.appName.widgetName")!
change the suite name to your own group.appBundle.widgetname (the group you created initially)

then inside your Live activity widget struct 
  let key = { (k: String) in context.attributes.prefixedKey(k) }

  let status = sharedDefault.string(forKey: key("status")) ?? "N/A"

why do we use the key? it's because the shared defaults adds like a timestamp or uuid to the key you send.


And if you'll need server updates make the contentState variables nullable
       // Nullable — only populated via APNs push.
       var status: String?

and then in the widget 
// Push update values take priority; fall back to UserDefaults (app-side updates)
//so it will accept both Flutter updates and updates from server.
let state = context.state
let status = state.status ?? sharedDefault.string(forKey: key("status")) ?? "N/A"
